Sources_v3/StreamChatUI/ChatChannel/MessageComposerInputAccessoryViewController.swift

```diff
@@ -170,40 +184,52 @@ open class MessageComposerInputAccessoryViewController<ExtraData: ExtraDataTypes
     // MARK: Button actions
     
     @objc func sendMessage() {
-        guard
-            let controller = controller,
-            let cid = controller.cid,
-            let text = composerView.messageInputView.textView.text,
+        guard let text = composerView.messageInputView.textView.text,
             !text.replacingOccurrences(of: " ", with: "").isEmpty
         else { return }
         
         switch state {
         case .initial:
             // TODO: Attachments
-            controller.createNewMessage(text: text)
-        case let .reply(messageToReply):
-            let messageController = controller.client.messageController(
-                cid: cid,
-                messageId: messageToReply.id
-            )
+            createNewMessage(text: text)
+        case .reply:
             // TODO:
             // 1. Attachments
             // 2. Should be inline reply after backend implementation.
-            messageController.createNewReply(text: textView.text, showReplyInChannel: true)
+            print("Inline reply sent.")
         case let .edit(messageToEdit):
-            let messageController = controller.client.messageController(
+            guard let cid = controller?.cid else { return }
+            let messageController = controller?.client.messageController(
                 cid: cid,
                 messageId: messageToEdit.id
             )
             // TODO: Adjust LLC to edit attachments also
-            messageController.editMessage(text: textView.text)
+            messageController?.editMessage(text: text)
         case let .slashCommand(command):
-            controller.createNewMessage(text: "/\(command.name) " + text)
+            createNewMessage(text: "/\(command.name) " + text)
         }
         
         state = .initial
     }
     
+    func createNewMessage(text: String) {
+        guard let cid = controller?.cid else { return }
```

Comment:
I see lot of those guards through the diff. Shouldn't we log somehow the issue when the values are nil when they shouldn't be? I mean this should never occur, so maybe use implicitly unwrapped optional in the declaration? It's a bit harder to debug while app doesn't at least `assert`-crash. What do you think, @VojtaStavik ?

Reply:
It was implicitly unwrapped before but there were some issue that it was accessed before setting. 
I can't remember what was that.

Reply:
ahh okay so lets leave it like this :)

Reply:
@DominikBucher12 has a good point here. Every silent failure like this is a potential hard-to-debug issue. Same below on L234 and L229. Someone will call `createNewMessage`, nothing will happen, and we will don't know there's a problem.

Let's add this to our guidelines - to avoid silent failures.

Reply:
Do we have some pattern for this cases already?